packages/api/src/controllers/taskController.js

```diff
@@ -520,11 +509,15 @@ const taskController = {
   },
 
   formatAnimalAndBatchIds(data) {
-    if (data.related_animal_ids) {
-      data.animals = [...new Set(data.related_animal_ids)].map((id) => ({ id }));
+    if ('related_animal_ids' in data) {
+      data.animals = data.related_animal_ids
+        ? [...new Set(data.related_animal_ids)].map((id) => ({ id }))
+        : null;
```

Comment:
If `related_animal_ids` is null (the intention is to remove the relation), set `data.animals` to null. The relation will be removed by sending `null` to `upsertGraph`.